pkg/controllermanager/controller/project/project_control.go

```diff
@@ -125,9 +127,11 @@ func (c *defaultControl) ReconcileProject(obj *gardenv1beta1.Project) (bool, err
 	if project.DeletionTimestamp != nil {
 		return c.delete(project, projectLogger)
 	}
-	if project.Generation != project.Status.ObservedGeneration {
+
+	if !apiequality.Semantic.DeepEqual(project.Status.Gardener, c.identity) || project.Generation != project.Status.ObservedGeneration {
```

Comment:
I would separate the conditions 
```suggestion
	if !apiequality.Semantic.DeepEqual(project.Status.Gardener, c.identity) {
		return false, c.reconcile(project, projectLogger)
	}
	
	if project.Generation != project.Status.ObservedGeneration {
			return false, c.reconcile(project, projectLogger)
	}
	
```